### PR TITLE
Create bucket only if allowed in region configuration

### PR DESCRIPTION
**Pull request by CChemin**

### CODE REVIEW

File: src/core/adapters/s3Client.ts [hunk 1]
```diff
@@ -283,10 +294,16 @@ export async function createS3Client(params: Params): Promise<S3Client> {
...
             const out: ReturnType<S3Client["list"]> = {
                 "directories": [],
-                "files": []
+                "files": [],
+                "errors": []
             };
 
             stream.once("end", () => dOut.resolve(out));
+            stream.on("error", e => {
+                console.log(e.message);
```
Comment (garronej) [inline src/core/adapters/s3Client.ts:303]: Forgotten console log

File: src/core/adapters/officialOnyxiaApiClient.ts [hunk 2]
```diff
@@ -251,6 +252,7 @@ export function createOfficialOnyxiaApiClient(params: {
                             const common: DeploymentRegion.S3.Common = {
                                 "monitoringUrlPattern": S3.monitoring?.URLPattern,
                                 "defaultDurationSeconds": S3.defaultDurationSeconds,
+                                "acceptBucketCreation": S3.acceptBucketCreation ?? true,
```
Comment (garronej) [inline src/core/adapters/officialOnyxiaApiClient.ts:255]: - To be grammatically correct it should be `acceptsBucketCreation` (with an s)
- Throughout the codebase all boolean start either by `is`, `does` or `do`.  

I suggest using `isBucketCreationAllowed`.  
Never mind if the API and the internal types diverge.

**Review by garronej — COMMENTED** [review]
Thank you for your work

File: src/core/adapters/s3Client.ts (hunk 1 above)
Comment (CChemin) [reply src/core/adapters/s3Client.ts:303]: don't you think this error can be useful to log ?

**Pull request closed without merging**